Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs

```diff
@@ -59,3 +74,41 @@ handler req = do
   logTagInfo "Payment completed : " ("bookingId " <> req.bookingId.getId <> ", rideId " <> req.rideId.getId)
 
 -- TODO store to DB
+
+handler EditLocationReq {..} = do
+  ride <- QRide.findById rideId >>= fromMaybeM (RideNotFound rideId.getId)
+  person <- QPerson.findById ride.driverId >>= fromMaybeM (PersonNotFound ride.driverId.getId)
+  whenJust origin $ \pickup -> do
+    startLocation <- buildLocation pickup
+    QL.create startLocation
+    pickupMapForBooking <- SLM.buildPickUpLocationMapping startLocation.id bookingId.getId DLM.BOOKING
+    QLM.create pickupMapForBooking
+    pickupMapForRide <- SLM.buildPickUpLocationMapping startLocation.id rideId.getId DLM.RIDE
+    QLM.create pickupMapForRide
+    let entityData = Notify.EditLocationReq {..}
+    Notify.notifyPickupOrDropLocationChange person.merchantId person.id person.deviceToken entityData
+
+buildLocation :: MonadFlow m => Common.Location -> m DL.Location
+buildLocation location = do
+  guid <- generateGUID
+  now <- getCurrentTime
+  return $
+    DL.Location
+      { id = guid,
+        createdAt = now,
+        updatedAt = now,
+        lat = location.gps.lat,
+        lon = location.gps.lon,
+        address =
+          DL.LocationAddress
+            { street = location.address.street,
+              door = location.address.door,
+              city = location.address.city,
+              state = location.address.state,
+              country = location.address.country,
+              building = location.address.building,
+              areaCode = location.address.area_code,
+              area = location.address.locality,
+              fullAddress = Nothing
```

Comment:
earlier also, are we sending fullAddress as Nothing ?
Can't we concat everything and construct fullAddress ?